core/src/main/java/nl/nn/adapterframework/doc/model/FrankElement.java

```diff
@@ -1,27 +1,209 @@
+	private final @Getter boolean isAbstract;
+	private @Getter boolean isDeprecated = false;
+
+	// Represents the Java superclass.
+	private @Getter FrankElement parent;
+
+	private Map<Class<? extends ElementChild>, LinkedHashMap<? extends AbstractKey, ? extends ElementChild>> allChildren;
+	private @Getter FrankElementStatistics statistics;
 
 	FrankElement(Class<?> clazz) {
-		this(clazz.getName(), clazz.getSimpleName());
+		this(clazz.getName(), clazz.getSimpleName(), Modifier.isAbstract(clazz.getModifiers()));
+		isDeprecated = clazz.getAnnotation(Deprecated.class) != null;
 	}
 
 	/**
 	 * Constructor for testing purposes. We want to test attribute construction in isolation,
 	 * in which case we do not have a parent.
+	 * TODO: Reorganize files such that this test constructor need not be public.
 	 */
-	FrankElement(final String fullName, final String simpleName) {
+	public FrankElement(final String fullName, final String simpleName, boolean isAbstract) {
 		this.fullName = fullName;
 		this.simpleName = simpleName;
+		this.isAbstract = isAbstract;
+		this.allChildren = new HashMap<>();
+		this.allChildren.put(FrankAttribute.class, new LinkedHashMap<>());
+		this.allChildren.put(ConfigChild.class, new LinkedHashMap<>());
+	}
+
+	public void setParent(FrankElement parent) {
+		this.parent = parent;
+		this.statistics = new FrankElementStatistics(this);
+	}
+
+	public void setAttributes(List<FrankAttribute> inputAttributes) {
+		setChildrenOfKind(inputAttributes, FrankAttribute.class);
+	}
+
+	private <C extends ElementChild> void setChildrenOfKind(List<C> inputChildren, Class<C> kind) {
+		LinkedHashMap<AbstractKey, C> children = new LinkedHashMap<>();
+		for(C c: inputChildren) {
+			if(children.containsKey(c.getKey())) {
+				log.warn(String.format("Frank element [%s] has multiple attributes / config children with key [%s]",
+						fullName, c.getKey().toString()));
+			} else {
+				children.put(c.getKey(), c);
+			}
+		}
+		allChildren.put(kind, children);
+	}
+
+	public List<FrankAttribute> getAttributes(Predicate<ElementChild> filter) {
+		return getChildrenOfKind(filter, FrankAttribute.class);
+	}
+
+	@SuppressWarnings("unchecked")
+	public <T extends ElementChild> List<T> getChildrenOfKind(
+			Predicate<ElementChild> selector, Class<T> kind) {
+		Map<? extends AbstractKey, ? extends ElementChild> lookup = allChildren.get(kind);
+		return lookup.values().stream().filter(selector).map(c -> (T) c).collect(Collectors.toList());
+	}
+
+	public void setConfigChildren(List<ConfigChild> children) {
+		setChildrenOfKind(children, ConfigChild.class);
+	}
+
+	public List<ConfigChild> getConfigChildren(Predicate<ElementChild> filter) {
+		return getChildrenOfKind(filter, ConfigChild.class);
+	}
+
+	ElementChild findElementChildMatch(ElementChild elementChild, Class<? extends ElementChild> kind) {
+		Map<? extends AbstractKey, ? extends ElementChild> lookup = allChildren.get(kind);
+		return lookup.get(elementChild.getKey());
+	}
```

Comment:
Kan je van ? een generic parameter maken?
```suggestion
	<C extends ElementChild> C findElementChildMatch(C elementChild) {
		Map<? extends AbstractKey, ? extends ElementChild> lookup = allChildren.get(elementChild.getClass());
		return lookup.get(elementChild.getKey());
	}
```

Reply:
Gedaan.